extensions/scarb-doc/theme/cairo.js

```diff
@@ -0,0 +1,246 @@
+    "i8",
+    "i16",
+    "i32",
+    "i64",
+    "i128",
+    "u8",
+    "u16",
+    "u32",
+    "u64",
+    "u128",
+    "usize",
+    "bool",
+    "Box",
+    "Option",
+    "Result",
+  ];
+  return {
+    name: "Cairo",
+    aliases: [ "cairo" ],
+    keywords: {
+      $pattern: hljs.IDENT_RE + "!?",
+      type: TYPES,
+      keyword: KEYWORDS,
+      literal: LITERALS,
+      built_in: BUILTINS,
+    },
+    illegal: "</",
+    contains: [
+      hljs.C_LINE_COMMENT_MODE,
+      hljs.COMMENT("/\\*", "\\*/", { contains: [ "self" ] }),
+      hljs.inherit(hljs.QUOTE_STRING_MODE, {
+        begin: /b?"/,
+        illegal: null,
+      }),
+      {
+        className: "string",
+        variants: [
+          { begin: /b?r(#*)"(.|\n)*?"\1(?!#)/ },
+          { begin: /b?'\\?(x\w{2}|u\w{4}|U\w{8}|.)'/ },
+        ],
+      },
+      {
+        className: "symbol",
+        begin: /'[a-zA-Z_][a-zA-Z0-9_]*/,
+      },
+      {
+        className: "number",
+        variants: [
+          { begin: "\\b0b([01_]+)" },
+          { begin: "\\b0o([0-7_]+)" },
+          { begin: "\\b0x([A-Fa-f0-9_]+)" },
+          { begin:
+              "\\b(\\d[\\d_]*(\\.[0-9_]+)?([eE][+-]?[0-9_]+)?)", },
+        ],
+        relevance: 0,
+      },
+      {
+        begin: [
+          /fn/,
+          /\s+/,
+          UNDERSCORE_IDENT_RE
+        ],
+        className: {
+          1: "keyword",
+          3: "title.function",
+        },
+      },
+      {
+        className: "meta",
+        begin: "#!?\\[",
+        end: "\\]",
+        contains: [
+          {
+            className: "string",
+            begin: /"/,
+            end: /"/,
+            contains: [ hljs.BACKSLASH_ESCAPE ],
+          },
+        ],
+      },
+      {
+        begin: [
+          /let/,
+          /\s+/,
+          /(?:mut\s+)?/,
+          UNDERSCORE_IDENT_RE
+        ],
+        className: {
+          1: "keyword",
+          3: "keyword",
+          4: "variable",
+        },
+      },
+      // must come before impl/for rule later
+      {
+        begin: [
+          /for/,
+          /\s+/,
+          UNDERSCORE_IDENT_RE,
+          /\s+/,
+          /in/
+        ],
+        className: {
+          1: "keyword",
+          3: "variable",
+          5: "keyword",
+        },
+      },
+      {
+        begin: [
+          /type/,
+          /\s+/,
+          UNDERSCORE_IDENT_RE
+        ],
+        className: {
+          1: "keyword",
+          3: "title.class",
+        },
+      },
+      {
+        begin: [
+          /(?:trait|enum|struct|union|impl|for)/,
+          /\s+/,
+          UNDERSCORE_IDENT_RE,
+        ],
+        className: {
+          1: "keyword",
+          3: "title.class",
+        },
+      },
+      {
+        begin: hljs.IDENT_RE + "::",
+        keywords: {
+          keyword: "Self",
+          built_in: BUILTINS,
+          type: TYPES,
+        },
+      },
+      {
+        className: "punctuation",
+        begin: "->",
```

Comment:
What is this one?

Reply:
this one is gonna be used in closures syntax:

https://github.com/starkware-libs/cairo/blob/106bb50f54e615ac7339e9ca1cb18743d0114579/crates/cairo-lang-semantic/src/expr/test_data/closure#L8